packages/lib/src/interactions/AxialSelectToZoom.tsx

```diff
@@ -0,0 +1,101 @@
+import { useThree } from '@react-three/fiber';
+import { Vector2 } from 'three';
+
+import { useCameraState } from '../vis/hooks';
+import { useAxisSystemContext } from '../vis/shared/AxisSystemProvider';
+import { getVisibleDomains } from '../vis/utils';
+import SelectionRect from './SelectionRect';
+import SelectionTool from './SelectionTool';
+import { useMoveCameraTo } from './hooks';
+import type { ModifierKey, Selection } from './models';
+import { getEnclosedRectangle } from './utils';
+
+interface Props {
+  axis: 'x' | 'y';
+  disabled?: boolean;
+  modifierKeys?: ModifierKey[];
+}
+
+function AxialSelectToZoom(props: Props) {
+  const { axis, modifierKeys, disabled } = props;
+
+  const { dataToWorld } = useAxisSystemContext();
+  const moveCameraTo = useMoveCameraTo();
+
+  const { width, height } = useThree((state) => state.size);
+  const camera = useThree((state) => state.camera);
+
+  const { xVisibleDomain, yVisibleDomain } = useCameraState(
+    getVisibleDomains,
+    []
+  );
+
+  function getAxialSelection(
+    mouseStartPoint: Vector2,
+    mouseEndPoint: Vector2
+  ): Selection {
+    const startPoint =
+      axis === 'x'
+        ? new Vector2(mouseStartPoint.x, yVisibleDomain[0])
+        : new Vector2(xVisibleDomain[0], mouseStartPoint.y);
+    const endPoint =
+      axis === 'x'
+        ? new Vector2(mouseEndPoint.x, yVisibleDomain[1])
+        : new Vector2(xVisibleDomain[1], mouseEndPoint.y);
+
+    return {
+      startPoint,
+      endPoint,
+      worldStartPoint: dataToWorld(startPoint),
+      worldEndPoint: dataToWorld(endPoint),
+    };
+  }
+
+  function onSelectionEnd(selection: Selection) {
+    const { startPoint, endPoint } = selection;
+
+    // Work in world coordinates as we need to act on the world camera
+    const { worldStartPoint, worldEndPoint } = getAxialSelection(
+      startPoint,
+      endPoint
+    );
+
+    if (
+      worldStartPoint.x === worldEndPoint.x &&
+      worldStartPoint.y === worldEndPoint.y
```

Comment:
Do we want to check only the coordinate that corresponds the `axis` prop here?

Reply:
It would make sense but I feel this makes the condition check more complex for the same result

Reply:
Well, if you draw a rectangle that is 100px tall (i.e. the height of the vis) but 0px wide and the `axis` is `x`, doesn't this mean that the code will try to divide by 0? Maybe replacing `&&` with `||` would do the trick?

Reply:
Oh, good thinking. Well, I can replace the `&&` by `||` and call it a day :smile: